code/modules/power/supermatter/supermatter.dm

```diff
@@ -1025,6 +1035,23 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	COOLDOWN_START(src, common_radio_cooldown, SUPERMATTER_COMMON_RADIO_DELAY)
 	return TRUE
 
+/obj/machinery/power/supermatter_crystal/proc/holiday_lights()
+	desc += "\nRadiating both festive cheer and actual radiation, it has a dazzling spectacle lights wrapped lovingly around the base transforming it from a potential doomsday device into a cosmic yuletide centerpiece."
+	holiday_lights = TRUE
+	RegisterSignal(src, COMSIG_ATOM_ATTACKBY, PROC_REF(holiday_attackby))
+	update_appearance()
+
+/obj/machinery/power/supermatter_crystal/proc/holiday_attackby(datum/source, obj/item/item, mob/living/user)
+	SIGNAL_HANDLER
+	if(istype(item, /obj/item/clothing/head/costume/santa))
+		balloon_alert(user, "placed hat")
+		QDEL_NULL(item)
+		desc += "\n\nThere's a santa hat placed atop it. How it got there without being dusted is a mystery."
```

Comment:
This should be checked for in `/examine` and not added onto the desc IMO.

Reply:
Currently, other than the overlay there's nothing to check in examine if there's a hat, since the object is deleted.

Reply:
I would either register a signal or set a flag

Reply:
Thanks, done.